src/manager.js

```diff
@@ -482,18 +515,47 @@ AdManager.prototype.loadAds = function(element, updateCorrelator) {
   }
 
   for (var i = 0; i < ads.length; i++) {
-    var thisEl = ads[i];
-
-    if ((thisEl.getAttribute('data-ad-load-state') === 'loaded') || (thisEl.getAttribute('data-ad-load-state') === 'loading')) {
+    var thisEl = ads[i],
+      slot,
+      adUnitConfig,
+      activeSizes,
+      gptSlotSizes,
+      adUnitSizes;
+
+	if ((thisEl.getAttribute('data-ad-load-state') === 'loaded') || (thisEl.getAttribute('data-ad-load-state') === 'loading')) {
       continue;
     }
 
-    var slot = this.configureAd(thisEl);
+    slot = this.configureAd(thisEl);
 
     if (slot && slot.eagerLoad) {
       slotsToLoad.push(slot);
     }
 
+    if (this.options.amazonEnabled) {
+ 
+    /**
+     * Try to use the gpt slot.getSizes method to retrieve the active sizes given the viewport parameters inside the ad config.
+     * This method is undocumented, and could be removed. When not available, fall back to all sizes specified in the ad unit itself.
+     * This is not optimal, as sizes which cannot be displayed due to the viewport dimensions will be requested from A9. It is thus used as a fallback.
+     * See Docs here https://developers.google.com/doubleclick-gpt/reference#googletagslot
+    */
+	
+      adUnitConfig = this.adUnits.units[thisEl.dataset.adUnit];
+      adUnitSizes = this.adUnitSizes(adUnitConfig.sizes)[1];
+
+      if (typeof gptSlotSizes == 'function') {
```

Comment:
Didn't notice this before but will this ever be a value other than `undefined` by the time it hits this block?  Seems like `gptSlotSizes` is just defined at the top of the function scope but never re-assigned.  Wondering why you couldn't re-assign that value for every slot you're iterating over, e.g.: 

```
gptSlotSizes = slot.getSizes();
activeSizes = this.adSlotSizes(gptSlotSizes);
```